dali/imgcodec/parsers/jpeg.cc

```diff
@@ -13,18 +13,65 @@
 // limitations under the License.
 
 #include "dali/imgcodec/parsers/jpeg.h"
+#include "dali/core/byte_io.h"
 
 namespace dali {
 namespace imgcodec {
 
+using jpeg_marker_t = std::array<uint8_t, 2>;
+
+constexpr jpeg_marker_t sos_marker = {0xff, 0xda};
+constexpr jpeg_marker_t soi_marker = {0xff, 0xd8};
+constexpr jpeg_marker_t eoi_marker = {0xff, 0xd9};
+
+bool isValidMarker(const jpeg_marker_t &marker) {
+  return marker[0] == 0xff && marker[1] != 0x00;
+}
+
+bool isSofMarker(const jpeg_marker_t &marker) {
+  if (!isValidMarker(marker) || marker[1] < 0xc0 || marker[1] > 0xcf) return false;
+  return (marker[1] != 0xc4 && marker[1] != 0xc8 && marker[1] != 0xcc);
+}
+
 ImageInfo JpegParser::GetInfo(ImageSource *encoded) const {
-  ImageInfo info;
-  // info.shape = {h, w, c};   // TODO(janton)
+  ImageInfo info{};
+  auto stream = encoded->Open();
+
+  jpeg_marker_t first_marker = stream->ReadOne<jpeg_marker_t>();
+  DALI_ENFORCE(first_marker == soi_marker);
+
+  bool read_shape = false;
+  while (!read_shape) {
+    auto marker = stream->ReadOne<jpeg_marker_t>();
+    if (marker[1] == 0xff) {
+      stream->SeekRead(-1, SEEK_CUR);
+      continue;
+    }
+
+    DALI_ENFORCE(isValidMarker(marker));
+    if (marker == sos_marker) break;
+
+    uint16_t size = ReadValueBE<uint16_t>(*stream);
+    ptrdiff_t next_marker_offset = stream->TellRead() - 2 + size;
```

Comment:
(just a suggestion) Alternatively, you could simply store the jump, and use SEEK_CUR in line 64.

Reply:
But between calculating `next_marker_offset` and doing the seek I (conditionally, if it's SOF) do some stream reads, so my position in the stream can change. That's why I think it's easier to use `SEEK_SET` here

Reply:
Fair enough